src/fluree/db/store.cljc

```diff
@@ -0,0 +1,74 @@
+(ns fluree.db.store
+  (:refer-clojure :exclude [read list exists?])
+  (:require [fluree.db.store.proto :as store-proto]
+            [fluree.db.store.file :as file-store]
+            [fluree.db.store.localstorage :as localstorage-store]
+            [fluree.db.store.memory :as mem-store]
+            [malli.core :as m]))
+
+
+(def BaseConfig
```

Comment:
Because it's data the schema can be composed into a larger config schema upstream to allow validation to happen at config ingestion:

```clojure
(def ConnectSchema
   {:storage/config store/StoreConfig
    :other/component component/ThatComponentConfig
    ;; etc.
  }
```
That would allow us to define the config relevant to storage right next to the storage, which I think is nice, but still use it wherever we want. That's what I like about malli!

Reply:
That needlessly couples code with configuration, which means any change to the format of the configuration file requires changes to the constructors that create the storage implementation and vice versa. it also spreads validation logic around the code that couples us to a particular validation library. 

Config validation should happen at the application boundary during initialization, and the same code that validates the config should then decompose the config and call the constructors as necessary. 

For example: there's no need for the `create-file-store` function to take a map. we should be able to initialize it with `(create-file-store "/some/path/")`, and the initialization code should know how to pull the right thing out of the config after it's validated and call the right function. this de-couples our configuration format from our internal application code. giving us more flexibility on both sides.